taxcalc/functions.py

```diff
@@ -184,11 +185,18 @@ def CapGains(p23250, p22250, _sep, ALD_Investment_ec, ALD_StudentLoan_HC,
     c23650 = p23250 + p22250
     # limitation on capital losses
     c01000 = max((-3000. / _sep), c23650)
-    # compute ymod* variables
+    # compute ymod1 variable that is included in AGI
     ymod1 = (e00200 + e00700 + e00800 + e00900 + e01400 + e01700 +
              (1 - ALD_Investment_ec) * (e00300 + e00600 +
                                         c01000 + e01100 + e01200) +
              e02000 + e02100 + e02300)
+    if CG_nodiff != 0.:
```

Comment:
@MattHJensen said:
> Some users do want to have both preferential rates and an exclusion. For example, the current House Republican Blueprint includes both preferential rates and a 50% exclusion. I believe @codykallen recently modeled this by just cutting the rates in half, but if we have a parameter called qdcg_exclusion, then some users might abuse it and try to implement it when CG_nodiff == 0. There is a warning in the notes about not doing that, but many users don't read the notes carefully and they won't receive any other warning that the reform they thought they were implementing wasn't implemented.
>
> If it adds significant complexity to relax the constraint that CG_nodiff!=0, then I understand taking the risk that some users make mistakes, but if it is simple to relax the constraint, then it is probably worthwhile for enhancing usability.

This pull request #1027 adds the `_CG_ec` parameter whose long name is:
```
Dollar amount of all capital gains and qualified dividends
that are excluded from AGI.
```
This makes sense when `_CG_nodiff` is not zero and all capital gains and qualified dividends are taxed at the same rates as other income.

But when `_CG_nodiff` is zero (as under current law), I don't know how to use the `_CG_ec` amount.  Don't we have to know what part of `_CG_ec` can be used to reduce taxable short-term capital gains as opposed to the part that can be used to reduce taxable long-term capital gains plus qualified dividends?  And how much of the part that reduces taxable LTCG+QDIV income can be used to reduce taxable special capital gains (like the gains on collectibles)?  Perhaps @codykallen has figured all this out, but there was nothing about this in the reform proposal that generated this pull request #1027.